ros_ign_bridge/CMakeLists.txt

```diff
@@ -84,6 +86,8 @@ foreach(PKG ${PACKAGES})
     PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 endforeach()
+target_link_libraries(ros_ign_interfaces_bridge builtin_interfaces_bridge)
```

Comment:
I wonder if it makes sense to include `builtin_interfaces` in the `bridge_utils` library above, since almost everything else is going to use it?

Reply:
There are a lot of interdependencies between interface packages, so I don't think adding `builtin_interfaces` to `bridge_utils` will help a lot.

e.g.: we didn't need to add a similar dependency (ros_ign_interfaces_bridge -> geometry_msgs_bridge) for this [line](https://github.com/ignitionrobotics/ros_ign/blob/6334294aac55a02158eb56d2f1505997db43fc55/ros_ign_bridge/src/convert/ros_ign_interfaces.cpp#L156). For some reason it works without it.

Why are we generating a `*_bridge` static library for each interface package?
We're only static linking them once into the dynamic library, so just adding all the source files to the dynamic library will result in cmake generating the same steps (i.e. if we're not using the static library more than once, there's no benefit of having them).

Reply:
> Why are we generating a *_bridge static library for each interface package?

It was a way to force compilation of each TU separately.  We were running up against memory/cpu limits any time another message definition was added in the deb builder stage.  We may be able to drop the static libraries at this point, but we needed to get it building for galactic/rolling syncs unblocked.

Reply:
I think those lines weren't needed, so removing them in https://github.com/ignitionrobotics/ros_ign/pull/216/commits/d0b6d99870a3ebfa71fbae1e62ff25d4a097065c.

---

> It was a way to force compilation of each TU separately

That's already the case when `add_library()` is used AFAIK. If an add_library() call has multiple source file, cmake will generate an object file for each one and then link everything together (checked one of the Makefiles cmake generates, that seems to be the case).
Maybe there's a subtle difference between both ways, I'm not sure.

If CI was running out of memory, I think we probably can modularize some of the header files that include ALL message files to avoid the issue.

Reply:
> Maybe there's a subtle difference between both ways, I'm not sure.

Probably not, this was probably a bit aggressive to try to clear the buildfarm.  I'll make a note to re-evaluate now that things are quieter and there isn't any sync pressure.